### DISCUSSION TITLE
Body class merges when navigating, setting to replace not working

### DISCUSSION THREAD

**Discussion by imaginarny** [nuxt/nuxt#23398] [Questions]
Let's say I have two pages, index and account. Index is a landing page styled differently from account. I don't use app.vue or layouts, simple pages are just enough.

Index has body class with tailwind styles, e.g. `bg-zinc-900`. Account has just `page-account` class. It is done so with `useHead()` by defining class in `bodyAttrs`.

When I navigate to `/account` from index page, account class is added/removed as I would wish. However, it also inherits the `bg-zinc-900` class from the index.

Reading [useHead docs](https://unhead.unjs.io/usage/guides/handling-duplicates#tagduplicatestrategy), it behaves like this by default, working sort of like attrs merging. This behavior although should be possible to change, by setting property `tagDuplicateStrategy` to `'replace'`. It doesn't do anything though and /account still inherits the index body class.

```js
useHead({
  bodyAttrs: {
    tagDuplicateStrategy: 'replace',
    class: '...'
  },
})
```

Is `tagDuplicateStrategy` not supported? Is anyone having the same issue? Feels strange no one is talking about it as it isn't something unusual to do. I'm also not sure I've noticed it before I updated Nuxt from v3.5.x to 3.7.3 earlier.

Ofc, I know I could style it the other way as a workaround, but for a sort of temporary coming soon page it made sense and never the less, this might be a bug.

**Comment by imaginarny** [comment]
Actually, I wasn't accurate about pages setup. While trying to replicate the issue, I've realized I'm using nested pages, as following:
```
pages/
├── account/
│   └──index.vue
├── account.vue
└── index.vue 
```

So `account.vue` contains `<NuxtPage />`. It looks like the culprit is the nested pages setup. Here is the replicated url:
https://stackblitz.com/edit/nuxt-starter-jzqgzm

Can anyone confirm the Nuxt behavior/logic constraints or unintended result of this setup?

Edit: sorry for tagging you @harlan-zw, but could you please take a look at this and confirm any of it? Whether it's Nuxt/Unhead issue or wrong use.

**Comment by harlan-zw** [comment]
Hey @imaginarny, thank you for the detailed investigation and reproduction.

I've spent some time looking into this but wasn't able to find the culprit, for some reason it looks like the side effects aren't being cleaned up.

Manually flushing them seems to work just need to figure out why Unhead isn't handling it itself.

```ts
onMounted(() => {
  Object.values(injectHead()._dom?.sideEffects || {}).forEach((fn) => fn());
});
```

Will try to revisit it when I have some time.

**Comment by imaginarny** [comment]
Interesting, thank you for the insight so far. 👍 This is something I wouldn't find out myself. 🤔
